File: 137singleNumberII/singleNumberII.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        auto [a, b]{std::array<int, 2>{}};
        std::for_each(std::cbegin(nums), std::cend(nums), [&](auto const num){
            b = (b ^ num) & ~a;
            a = (a ^ num) & ~b;});
        return b;
    }
};

File: 137singleNumberII/singleNumberII.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    let [a, b] = Array(2).fill(0)
    nums.forEach(num => {
        b = (b ^ num) & ~a
        a = (a ^ num) & ~b})
    return b
};
